driver-core/src/main/com/mongodb/TaggableReadPreference.java

```diff
@@ -52,17 +53,21 @@
 
     private final List<TagSet> tagSetList = new ArrayList<TagSet>();
     private final Long maxStalenessMS;
+    private final ReadPreferenceHedgeOptions hedgeOptions;
```

Comment:
One final change with respect to the new "with" methods.  Since all the subclasses of this class are non-public, it doesn't help so much to make their overrides return TaggableReadPreference.  But I think you can make it work by redeclaring the "with" methods on this class (which is public) with that return type.  For example:
   
   @Override
   public abstract TaggableReadPreference withTagSet(TagSet tagSet);